include/seqan3/range/view/detail.hpp

```diff
@@ -60,6 +60,26 @@ namespace seqan3::detail
 struct view_base : public ranges::view_base
 {};
 
+// ============================================================================
+//  size_type_t_or_void
+// ============================================================================
+
+//!\brief TransformationTrait that resolves to seqan3::size_type_t if possible and void otherwise. [Default Overload]
+template <typename t>
+struct size_type_t_or_void
+{
+    //!\brief The default is to expose `void`.
+    using type = void;
+};
+
+//!\brief TransformationTrait that resolves to seqan3::size_type_t if possible and void otherwise. [Specialisation]
+template <std::ranges::SizedRange t>
+struct size_type_t_or_void<t>
+{
+    //!\brief For sized ranges the seqan3::size_type_t is exposed.
+    using type = size_type_t<t>;
```

Comment:
there is a size_type_t mf? Does this wrap some ranges code or is it our own?
Just curious.